### DISCUSSION TITLE
Staronové features

### DISCUSSION THREAD

**Discussion by arozehnal** [JiriBilek/geokuk#2] [Ideas]
Jsem strašně rád, že se GK někdo ujal. Původní autor se mi ozval cca před rokem s tím, že se GK nemůže moc věnovat. Používám GK od jeho počátků a v zásadě mi v něm nic nechybí.
Ale měl bych náměty, které by GK podle mého pomohly k "nesmrtelnosti":

1. pokud by uměl znovu pracovat s OSM, ať už online nebo offline (myslím https://osm.paws.cz/  )
2. zrevidovat funkčnost stahování map

Tím otevřením se OSM by se GK zbavil smrtelné závislosti na Seznam mapách. Pokud mohu volit, tak sáhnu po seznamových, pokud ale nemohu volit, tak se spokojím i s OSM.

U toho stahování stále nějak funguje průběžné stahování při brouzdání mapou. Soubor s kachlemi mi roste a když Seznam mapy bloknul, tak jsem zjistil, že tato keš není špatná. Problém je v tom, jak efektivně vychytat nestažené dlaždice. Vzpomínám na počátky GK, kdy GK uměl cíleně postahovat zadanou oblast. V době, kdy nebylo tak dostupné wifi a signál vůbec, to bylo k nezaplacení.

**Comment by JiriBilek** [comment]
Ten kód je dost složitý, takže se snažím dělat změny, které nejsou strukturální.

Ty OSM online by měly jít, u offline by to bylo asi moc práce. Ještě by se daly rozchodit google mapy, protože jsem před časem přidal google ortofoto (seznam v cizině to měl bídné), ale tam jsme licenčně opět v problému.

Stahování oblasti nevím, jak funguje, ale co vyzkoušet udělat tisk? Ten jde do souboru, který si smažeš, takže no stress. A pro tisk musí Geokuk dlaždice postahovat. 
Vnitřní keš Geokuku je spolehlivá, takhle jsem fungoval v cizině, kde jsem neměl data. Doma jsem ručně postahoval a bylo.

Ještě info: necommituju do **main**u, ale do větve ~~lokalni_vetev_produkce~~ **lokalni_upravy_5.1**.

**Comment by arozehnal** [comment]
Jasně, Do žádných velkých změn tě netlačím, Těch Google map jsem si už všiml a to mě vlastně vybavilo  dvě(doufám) drobnosti...
v kontextové nabídce aktuálně umístěného kurzoru je "Otevřít Streetview" nešlo by přidat seznamovou Panoramu?

Ve stejné nabídce je Přidat do cesty resp. Souřadnice do clipboardu. Obojí často používám.
Někdy by se mi hodilo pouhé naklikání vícero bodů, ale pouze samostatně mimo cestu a trvale s umístěním do souboru - tj taková analogie s uložení cest nebo označení bodů do kategorie typu Lovím. Úplně by mi stačilo to nejjednodušší - co řádek - to souřadnice bodu. Soubor s body už zpět do GK nepotřebuji načítat.
A dík za ten tip s tiskem příležitostně to zkusím.

**Reply by JiriBilek** [reply]
Ahoj,
přidal jsem Openstreetmap, release je na https://github.com/JiriBilek/geokuk/blob/zmena_map_5.1/releases/geokuk-5.1.1a.jar

**Reply by arozehnal** [reply]
Díky.
